### ISSUE TITLE
Set up testing environment

### ISSUE THREAD

**Issue by ricardogsilva** [kartoza/ckanext-dalrrd-emc-dcpr#54]
We need to set up a **testing :orange_circle:** environment for this project.

It shall use Kartoza's existing k8s cluster, which is based on Linux hosts.

The existing `docker/docker-compose.yml` and `docker/docker-compose.ci.yml` files are a good starting point for building the appropriate helm chart. 

Note that we are going to need to use additional custom ckan configuration files for each deployment environment, which means we need one for **testing** too - This configuration file includes a number of sensitive details and therefore must be treated as a secret. It shall be named `ckan-testing-settings.ini`. You may use the `docker/ckan-ci-settings.ini` as an example, but please note:

- The `ckan-ci-settings.ini` file inherits some values from the `docker/ckan-dev-settings.ini` file - This means that this new `ckan-testing-settings.ini` file should also inherit from the `ckan-dev-settings.ini` file.

- The settings file should redefine all settings that use sensitive information, namely:

    - [ ] Be sure to set the **debug = false** parameter in order to avoid leaking potentially sensitive information to the public
    - [ ] `beaker.session.secret`
    - [ ] `app_instance_uuid` - this one is not strictly a secret, but it should be regenerated
    - [ ] `sqlalchemy.url`
    - [ ] `ckan.datastore.write_url`
    - [ ] `ckan.datastore.read_url`
    - [ ] `ckan.site_url`
    - [ ] `api_token.jwt.encode.secret`
    - [ ] `api_token.jwt.decode.secret`
    - [ ] `solr_url`
    - [ ] `ckan.redis.url`
    - [ ] `ckan.harvest.mq.type`
    - [ ] `ckan.harvest.mq.hostname`
    - [ ] `ckan.harvest.mq.port`
    - [ ] `ckan.harvest.mq.redis_db`
    - [ ] `ckan.site_title` - set this to `SASDI EMC-DCPR Testing`
    - [ ] `email_to`
    - [ ] `error_email_from`
    - [ ] `smtp.server`
    - [ ] `smtp.starttls`
    - [ ] `smtp.user`
    - [ ] `smtp.password`
    - [ ] `smtp.mail_from`
    - [ ] `smtp.reply_to`

--- 

### Automated deployment

Every time we do a commit to this repo's `main` branch, the corresponding docker image is built and pushed to the registry at [kartoza/ckanext-dalrrd-emc-dcpr](https://hub.docker.com/r/kartoza/ckanext-dalrrd-emc-dcpr/tags) (with a `main` tag). Ideally we would be able to automatically deploy the image into the **testing** environment as soon as it becomes available on the docker registry. I'm not sure this is still appropriate for the scope of this issue, if not we can leave it for another issue - please comment on this.

**Comment by lucernae** [comment]
Hi @ricardogsilva , I've tried to understand how `datastore-db` service is being used. I don't see it being mentioned in the config files at all. Does that mean I just need to spin up separate db container without any preconfigured setup besides user/password?

**Comment by ricardogsilva** [comment]
The `datastore-db` is still not being used because we are waiting on #13 for that.

For the moment yes, spin up an additional DB container with a single user

**Comment by ricardogsilva** [comment]
@lucernae please comment on the above statements about automated deployments

**Comment by lucernae** [comment]
We are not able to do that yet, but it is possible.
Currently the app configuration uses file as application parameter (in ckan.ini and who.ini).
This forces me to use custom tooling of helm+kustomize (had to be both not just one). ArgoCD Image Updater plugin only supports either helm or kustomize but not both.
I have some plan to allow that by modifying the ArgoCD docker image, but it's not ready yet.

**Comment by lucernae** [comment]
@ricardogsilva since most of the configuration are privates for now, I put all the manifests on [kartoza/devops](https://github.com/kartoza/devops/tree/master/rancher-2.x/manifests/projects/emc).

I will create separate tickets for the refactor plan so that extract the generic manifests in this repo and only stores the private ones in kartoza/devops repo.

**Comment by ricardogsilva** [comment]
Testing environment baseline setup is done. It can be consulted at

https://testing.emc.kartoza.com

Closing this issue
